dashboard/app/controllers/projects_controller.rb

```diff
@@ -24,11 +24,12 @@ class ProjectsController < ApplicationController
   def index
   end
 
-  def template
-    # sanitize user input by whitelisting templates we are willing to render
-    head :not_found and return unless TEMPLATES.include? params[:template]
+  def redirect_legacy
```

Comment:
It looks like if I go to an old share page (i.e. http://localhost:3000/p/applab#M17JvqZ68AlDVpRfsdUSSw), sign in is required, which I don't think is the spec'ed behavior. Maybe we can get away with adding this as an authenticate_user exception? I think that's okay, because then we'll load the page, do a hard JS redirect, and authenticate or not authenticate appropriately for the redirected URL.